bundles/runtime/org.eclipse.fx.ui.controls/src/org/eclipse/fx/ui/controls/tabpane/DndTabPane.java

```diff
@@ -0,0 +1,153 @@
+package org.eclipse.fx.ui.controls.tabpane;
+
+import static org.eclipse.fx.ui.controls.tabpane.DndTabPaneFactory.setup;
+import static org.eclipse.fx.ui.controls.tabpane.DndTabPaneFactory.teardown;
+
+import java.util.function.Consumer;
+
+import org.eclipse.fx.ui.controls.tabpane.DndTabPaneFactory.DragSetup;
+import org.eclipse.fx.ui.controls.tabpane.DndTabPaneFactory.FeedbackType;
+import org.eclipse.fx.ui.controls.tabpane.skin.DnDTabPaneSkinHookerFullDrag;
+import org.eclipse.fx.ui.controls.tabpane.skin.DndTabPaneSkinHooker;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.scene.control.Skin;
+import javafx.scene.control.SkinBase;
+import javafx.scene.control.TabPane;
+
+/**
+ * A TabPane with Drag and Drop Support
+ *
+ * @author Adrodoc55
+ */
+public class DndTabPane extends TabPane {
+	private static final FeedbackType DEFAULT_FEEDBACK_TYPE = FeedbackType.MARKER;
+	private final BooleanProperty allowDetach = new SimpleBooleanProperty();
+	private final ObjectProperty<FeedbackType> feedbackType = new SimpleObjectProperty<>(DEFAULT_FEEDBACK_TYPE);
```

Comment:
please pass `this` to as the first argument to `new SimpleObjectProperty<>` we could even think of lazy property initialization for feedbackType